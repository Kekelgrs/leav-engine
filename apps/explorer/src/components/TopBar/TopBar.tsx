// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {BellOutlined, MenuOutlined, SelectOutlined} from '@ant-design/icons';
import React from 'react';
import {useTranslation} from 'react-i18next';
import styled from 'styled-components';
import themingVars from '../../themingVar';
import HeaderNotification from '../HeaderNotification';
import UserMenu from './UserMenu';
interface ITopBarProps {
    sideBarVisible: boolean;
    userPanelVisible: boolean;
    toggleSidebarVisible: () => void;
    toggleUserPanelVisible: () => void;
}

const Wrapper = styled.div`
    display: grid;
    grid-template-rows: 5rem;
    grid-template-columns: 5rem 1fr repeat(2, 8rem) 10rem;
    align-items: start;
    justify-items: start;
    justify-content: start;

    color: white;
    background: transparent linear-gradient(85deg, #0f2027 0%, #203a43 52%, #2c5364 100%) 0% 0% no-repeat padding-box;
`;
interface IMenuItemProps {
    isActive?: boolean;
}
const MenuItem = styled.div<IMenuItemProps>`
    display: flex;
    justify-content: space-around;
    align-items: center;
    transition: 275ms ease-out;
    padding: 0 1rem;
    background: ${props => (props.isActive ? themingVars['@primary-color'] : 'none')};

    & > * {
        margin: 0 0.5rem;
    }

    &:hover {
        background: ${themingVars['@primary-color']};
    }
`;

const MenuItemUser = styled(MenuItem)`
    justify-self: end;
`;

const WrapperHeaderNotification = styled.div`
    width: 100%;
    display: flex;
    justify-content: start;
    align-items: center;
`;

function TopBar({
    sideBarVisible,
    userPanelVisible,
    toggleSidebarVisible,
    toggleUserPanelVisible
}: ITopBarProps): JSX.Element {
    const {t} = useTranslation();

    const triggerMenuKey = 'trigger-side-menu';
    const userPanelKey = 'trigger-user-panel';

    const handleSideMenuClick = () => {
        toggleSidebarVisible();
    };

    const handleUserPanelClick = () => {
        toggleUserPanelVisible();
    };

    return (
        <Wrapper>
            <MenuItem key={triggerMenuKey} onClick={handleSideMenuClick} isActive={sideBarVisible}>
                <MenuOutlined />
            </MenuItem>

            <WrapperHeaderNotification key="lib-name">
                <HeaderNotification />
            </WrapperHeaderNotification>

            <MenuItem key="shortcuts">
                <SelectOutlined />
                {t('menu.shortcuts')}
            </MenuItem>

            <MenuItem key="event">
                <BellOutlined />
                {t('menu.events')}
            </MenuItem>

            <MenuItemUser key={userPanelKey} onClick={handleUserPanelClick} isActive={userPanelVisible}>
                <UserMenu />
            </MenuItemUser>
        </Wrapper>
    );
}

export default TopBar;
