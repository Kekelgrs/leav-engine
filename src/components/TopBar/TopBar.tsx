import React from 'react';
import {Input, Menu} from 'semantic-ui-react';

interface ITopBarProps {
    toggleSidebarVisible: () => void;
}

function TopBar({toggleSidebarVisible}: ITopBarProps): JSX.Element {
    return (
        <Menu inverted style={{borderRadius: 0, margin: 0}}>
            <Menu.Menu>
                <Menu.Item name="toggle-sidebar" icon="sidebar" content="" onClick={toggleSidebarVisible} />
                <Menu.Item name="Library name" />
            </Menu.Menu>
            <Menu.Menu position="right">
                <Menu.Item>
                    <Input icon="search" placeholder="Search..." />
                </Menu.Item>
            </Menu.Menu>
        </Menu>
    );
}

export default TopBar;
