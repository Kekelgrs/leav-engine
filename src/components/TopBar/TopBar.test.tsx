import React from 'react';
import {render} from 'enzyme';
import TopBar from './TopBar';

describe('TopBar', () => {
    test('Snapshot test', async () => {
        const comp = render(<TopBar />);

        expect(comp).toMatchSnapshot();
    });
});
