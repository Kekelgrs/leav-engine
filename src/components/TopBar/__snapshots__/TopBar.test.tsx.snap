// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar Snapshot test 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ui inverted menu",
      "style": "border-radius:0;margin:0",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "menu",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "item",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "sidebar icon",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "item",
              },
              "children": Array [
                Object {
                  "data": "Library Name",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "item",
            },
            "children": Array [
              Object {
                "data": "Library Name",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "item",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "sidebar icon",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "right menu",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "item",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "ui icon input",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "placeholder": "Search...",
                        "type": "text",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "attribs": Object {
                          "aria-hidden": "true",
                          "class": "search icon",
                        },
                        "children": Array [],
                        "name": "i",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "aria-hidden": undefined,
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "aria-hidden": undefined,
                          "class": undefined,
                        },
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "placeholder": undefined,
                        "type": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "placeholder": undefined,
                        "type": undefined,
                      },
                    },
                    Object {
                      "attribs": Object {
                        "aria-hidden": "true",
                        "class": "search icon",
                      },
                      "children": Array [],
                      "name": "i",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "placeholder": "Search...",
                          "type": "text",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "placeholder": undefined,
                          "type": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "placeholder": undefined,
                          "type": undefined,
                        },
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "aria-hidden": undefined,
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "aria-hidden": undefined,
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "right menu",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "item",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "ui icon input",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "placeholder": "Search...",
                      "type": "text",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "attribs": Object {
                        "aria-hidden": "true",
                        "class": "search icon",
                      },
                      "children": Array [],
                      "name": "i",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "aria-hidden": undefined,
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "aria-hidden": undefined,
                        "class": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "placeholder": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "placeholder": undefined,
                      "type": undefined,
                    },
                  },
                  Object {
                    "attribs": Object {
                      "aria-hidden": "true",
                      "class": "search icon",
                    },
                    "children": Array [],
                    "name": "i",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "placeholder": "Search...",
                        "type": "text",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "placeholder": undefined,
                        "type": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "placeholder": undefined,
                        "type": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "aria-hidden": undefined,
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "aria-hidden": undefined,
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "menu",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "item",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "sidebar icon",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "item",
                },
                "children": Array [
                  Object {
                    "data": "Library Name",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "item",
              },
              "children": Array [
                Object {
                  "data": "Library Name",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "item",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "aria-hidden": "true",
                      "class": "sidebar icon",
                    },
                    "children": Array [],
                    "name": "i",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "aria-hidden": undefined,
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "aria-hidden": undefined,
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
